framework/test/unit/modules/interoperability/utils.spec.ts

```diff
@@ -68,4 +104,598 @@ describe('Utils', () => {
+			expect(partnerChannelStoreMock.setWithSchema).toHaveBeenCalledTimes(1);
+			expect(context.getStore).toHaveBeenCalledTimes(1);
+			expect(partnerValidators.certificateThreshold).toEqual(params.newCertificateThreshold);
+			expect(partnerChannelData.partnerChainOutboxRoot).toEqual(
+				defaultCalculatedRootFromRightWitness,
+			);
+		});
+
+		it('should run successfully and return undefined when newCertificateThreshold is zero', async () => {
+			const paramsWithThresholdZero = {
+				activeValidatorsUpdate,
+				newCertificateThreshold: BigInt(0),
+				certificate: Buffer.alloc(2),
+				inboxUpdate,
+			};
+			const contextWithThresholdZero: any = { ...context, params: paramsWithThresholdZero };
+
+			await expect(
+				commonCCUExecutelogic({
+					certificate,
+					partnerChainAccount,
+					partnerValidatorStore,
+					partnerChainStore,
+					partnerValidators,
+					chainIDBuffer,
+					context: contextWithThresholdZero,
+				}),
+			).resolves.toBeUndefined();
+			expect(partnerValidatorStore.setWithSchema).toHaveBeenCalledTimes(1);
+			expect(partnerChainStore.setWithSchema).toHaveBeenCalledTimes(1);
+			expect(partnerChannelStoreMock.getWithSchema).toHaveBeenCalledTimes(1);
+			expect(partnerChannelStoreMock.setWithSchema).toHaveBeenCalledTimes(1);
+			expect(context.getStore).toHaveBeenCalledTimes(1);
+			expect(partnerValidators.certificateThreshold).toEqual(BigInt(12)); // original partnerValidator value unchanged
+			expect(partnerChannelData.partnerChainOutboxRoot).toEqual(
+				defaultCalculatedRootFromRightWitness,
+			);
+		});
+
+		it('should run successfully and return undefined when certificate is empty', async () => {
+			const paramsWithEmptyCertificate = {
+				activeValidatorsUpdate,
+				newCertificateThreshold: params.newCertificateThreshold,
+				certificate: EMPTY_BYTES,
+				inboxUpdate,
+			};
+			const contextWithEmptyCertificate: any = { ...context, params: paramsWithEmptyCertificate };
+
+			await expect(
+				commonCCUExecutelogic({
+					certificate: {} as any,
+					partnerChainAccount,
+					partnerValidatorStore,
+					partnerChainStore,
+					partnerValidators,
+					chainIDBuffer,
+					context: contextWithEmptyCertificate,
+				}),
+			).resolves.toBeUndefined();
+			expect(partnerValidatorStore.setWithSchema).toHaveBeenCalledTimes(1);
+			expect(partnerChainStore.setWithSchema).not.toHaveBeenCalled();
+			expect(partnerChannelStoreMock.getWithSchema).toHaveBeenCalledTimes(1);
+			expect(partnerChannelStoreMock.setWithSchema).toHaveBeenCalledTimes(1);
+			expect(context.getStore).toHaveBeenCalledTimes(1);
+			expect(partnerChainAccount.lastCertificate.height).toEqual(5); // original partnerValidator value unchange
+			expect(partnerChannelData.partnerChainOutboxRoot).toEqual(
+				defaultCalculatedRootFromRightWitness,
+			);
+		});
+
+		it('should run successfully and return undefined when messageWitness is empty', async () => {
```

Comment:
maybe also check if `calculateRootFromRightWitness` is used when message witness is not empty and vise versa